plugins/amazonq/shared/jetbrains-community/src/software/aws/toolkits/jetbrains/services/amazonq/profile/QRegionProfileManager.kt

```diff
@@ -59,16 +58,24 @@ class QRegionProfileManager : PersistentStateComponent<QProfileState>, Disposabl
             )
     }
 
-    // should be call on project startup to validate if profile is still active
+    /**
+     * Called on project startup to validate if selected profile is still active
+     */
+    @Deprecated("This is a giant hack and we are not handling all the cases")
     @RequiresBackgroundThread
     fun validateProfile(project: Project) {
         val conn = getIdcConnectionOrNull(project)
         val selected = activeProfile(project) ?: return
-        val profiles = tryOrNull {
+        val profiles = try {
             listRegionProfiles(project)
+        } catch (_: Exception) {
```

Comment:
Can we catch network exceptions here?

Reply:
i think the only exception acceptable to not catch is if they are not allowed to use Q